library/net.c

```diff
@@ -1193,6 +1193,8 @@ struct dnet_net_state *dnet_state_create(struct dnet_node *n,
 
 err_out_send_destroy:
 	dnet_state_reset(st, err);
+	if (atomic_read(&st->refcnt) > 1)
```

Comment:
What did creates this extra reference to the state? Could it be done more accurate?